integrationtests/self/http_test.go

```diff
@@ -1020,4 +1030,19 @@ var _ = Describe("HTTP tests", func() {
 			Expect(num0RTTPackets.Load()).To(BeNumerically(">", 0))
 		})
 	})
+
+	It("sends and receives trailers", func() {
+		resp, err := client.Get(fmt.Sprintf("https://localhost:%d/trailers", port))
+		Expect(err).ToNot(HaveOccurred())
+		Expect(resp.StatusCode).To(Equal(200))
+		body, err := io.ReadAll(gbytes.TimeoutReader(resp.Body, 3*time.Second))
+		Expect(err).ToNot(HaveOccurred())
+		Expect(string(body)).To(Equal("Hello, World!\n"))
+		Expect(resp.Header).To(Not(HaveKey("Grpc-Status")))
+		Expect(resp.Header).To(Not(HaveKey("Grpc-Message")))
+		Expect(resp.Header).To(Not(HaveKey("Grpc-Details")))
+		Expect(resp.Header).To(Not(HaveKey("Trailer:Grpc-Details")))
```

Comment:
I added this test. I didn't have this before because it definitely explodes on the client side because colon is definitely not a valid character for headers. In either case, it's still good to have an explicit check. 👍